coreplugins/cesiumion/public/components/TasksDialog.jsx

```diff
@@ -0,0 +1,183 @@
+
+const TaskStatusItem = ({
+    asset,
+    progress,
+    task,
+    helpText = '',
+    active = true,
+    bsStyle = 'primary'
+}) => (
+    <div className="list-group-item">
+        <div className="row">
+            <div className="col-xs-6">
+                <p style={{ fontWeight: 'bold' }}>
+                    <IonAssetLabel asset={asset} showIcon={true} />
+                </p>
+            </div>
+            <div className="col-xs-6">
+                <p className="pull-right">Status: {task}</p>
+            </div>
+        </div>
+        <progress value={progress} max="100" className={bsStyle}></progress>
+        {helpText && <small>{helpText}</small>}
+    </div>
+);
+
+export default class TaskDialog extends React.Component {
+    static defaultProps = {
+        tasks: [],
+        taskComponent: TaskStatusItem,
+        show: false
+    };
+
+    static propTypes = {
+        tasks: PropTypes.array.isRequired,
+        taskComponent: PropTypes.elementType,
+        onClearFailed: PropTypes.func.isRequired,
+        onHide: PropTypes.func.isRequired,
+        show: PropTypes.bool
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            showModal: props.show,
+            error: ''
+        };
+
+        this.setModal = this.setModal.bind(this);
+        this.show = this.show.bind(this);
+        this.hide = this.hide.bind(this);
+    }
+
+    setModal(domNode) {
+        this.modal = domNode;
+    }
+
+    componentDidMount() {
+        this._mounted = true;
+
+        $(this.modal)
+            .on('hidden.bs.modal', () => {
+                this.hide();
+            })
+            .on('shown.bs.modal', () => {
+                if (this.props.onShow) this.props.onShow();
+            });
+
+        this.componentDidUpdate();
+    }
+
+    componentWillUnmount() {
+        this._mounted = false;
+        $(this.modal).off('hidden.bs.modal shown.bs.modal').modal('hide');
+    }
+
+    componentDidUpdate() {
+        if (this.state.showModal) {
+            $(this.modal).modal('show');
+        } else {
+            $(this.modal).modal('hide');
+        }
+    }
+
+    show() {
+        this.setState({ showModal: true, error: '' });
+    }
+
+    hide() {
+        this.setState({ showModal: false });
+        if (this.props.onHide) this.props.onHide();
+    }
+
+    renderTaskItems() {
+        const { tasks, taskComponent: TaskComponent } = this.props;
+        let hasErrors = false;
+
+        const taskItems = tasks.map(
+            ({ type: asset, upload, process, error }) => {
+                let task,
+                    style,
+                    progress = 0;
+
+                if (upload.active) {
+                    progress = upload.progress;
+                    task = 'Uploading';
+                    style = 'info';
+                } else if (process.active) {
+                    progress = process.progress;
+                    task = 'Processing';
```

Comment:
Is the `task` variable a key or display string or both?

Reply:
From what I understand it is a map of `TaskComponents` where each task has a descriptive string value depending on the status of the async upload.